thavalon-webapp/src/utils/accountManager.tsx

```diff
@@ -0,0 +1,251 @@
+        }
+    }
+    /**
+     * Queries the server to see if refresh token we currently have is valid.
+     */
+    private async checkRefreshToken(): Promise<HttpResponse> {
+        console.log("Checking refresh token!");
+        let httpResponse: HttpResponse = {
+            "result": true,
+            "message": "",
+        }
+        // returns 500 or 401 or 200
+        return await fetch("/api/auth/refresh", {
+            method: "POST",
+            credentials: "include"
+        }).then((response) => {
+            if (response.status === STATUS_OK) {
+                response.json().then((jwt: JwtType) => {
+                    // use anonymous function to get around this being unbound
+                    this.setJwtInfo(jwt, () => (this.checkRefreshToken()));
+                });
+            } else if (response.status === STATUS_UNAUTHORIZED) {
+                httpResponse.result = false;
+                httpResponse.message = "Unauthorized when trying to refresh token";
+            } else if (response.status === STATUS_INTERNAL_SERVER_ERROR) {
+                httpResponse.result = false;
+                httpResponse.message = "Internal server error when refreshing token";
+            } else {
+                httpResponse.result = false;
+                httpResponse.message = "Unexpected return code from server: " + response.status;
+            }
+            return httpResponse;
+        }).catch((error) => {
+            console.log("Failed to refresh token, error is: " + error);
+            httpResponse.result = false;
+            httpResponse.message = "Unable to refresh token, try again";
+            return httpResponse;
+        });
+    }
+
+    public async checkLoggedIn(): Promise<HttpResponse> {
+        return await this.checkRefreshToken();
+    }
+
+    /**
+     * Registers a user with the given info.
+     * 
+     * @param name The name of the user
+     * @param email The email address of the user
+     * @param password The password of the user
+     * @returns HttpResponse with result (true on success, false otherwise) and message set if result is false
+     */
+    public async registerUser(name: string, email: string, password: string): Promise<HttpResponse> {
+        // parameters for registering user
+        let addUserInfo: AddUserInfo = {
+            "displayName": name,
+            "email": email,
+            "password": password
+        }
+
+        let httpResponse: HttpResponse = {
+            "result": true,
+            "message": ""
+        }
+        // Following end point can return 201 on successful add or 406 on reject or 500 if everything's broken or 409 if duplicate account
+        return await fetch("/api/add/user", {
+            method: "POST",
+            body: JSON.stringify(addUserInfo),
+            headers: {
+                "Content-Type": "application/json"
+            },
+            credentials: "include"
```

Comment:
We can remove `credentials: "include"` here, since we don't need the refresh token to send.